e2e/auth.setup.ts

```diff
@@ -0,0 +1,110 @@
+// Required environment variables:
+// - ADMIN_USERNAME: username for admin
+// - ADMIN_PASSWORD: password for admin
+//
+// Environment variables:
+// - in CI: set in repo GitHub secrets, read via env in workflow
+// - locally: set in .env file, read by dotenv
+
+import { Page, expect, test as setup } from '@playwright/test'
+import { v4 as uuidv4 } from 'uuid'
+import { ADMIN_STORAGE_STATE, USER_STORAGE_STATE } from '../playwright.config'
+import { getEndpoint } from './helpers/http'
+import { setLocalStorage } from './helpers/localStorage'
+import {
+  USER_NAME_LOCALSTORAGE_KEY,
+  createTestUser,
+  deleteTestUser,
+  getAccessTokenFromCookie,
+  getAdminUserCredentials,
+  loginTestUser,
+} from './helpers/testUser'
+import { TestUser } from './helpers/types'
+
+let testUserId: string | undefined
+let adminAccessToken: string | undefined
+let adminPage: Page
+let userPage: Page
+
+setup.describe('Setup', () => {
+  setup.beforeAll(async ({ browser }) => {
+    adminPage = await browser.newPage()
+    userPage = await browser.newPage()
+  })
+
+  setup('authenticate users', async () => {
+    setup.slow()
+
+    await setup.step('authenticate admin', async () => {
+      const { adminUserName, adminUserPassword } = getAdminUserCredentials()
+
+      await loginTestUser(adminPage, adminUserName!, adminUserPassword!)
+      await adminPage.context().storageState({ path: ADMIN_STORAGE_STATE })
```

Comment:
Store the [authenticated state](https://playwright.dev/docs/auth#basic-shared-account-in-all-tests), so that it can be reused across tests.